app/Concerns/HasAuthorsAndEditors.php

```diff
@@ -41,6 +42,16 @@ public function withUsers(array|Entity|null $records): array|Entity|null
         array_walk($records, static function (&$post) use ($users) {
             $post->author = $users[$post->author_id] ?? null;
             $post->editor = $users[$post->editor_id] ?? null;
+
+            // Get email to grab avatar later
+            if ($identityAuthor = $post->author?->getIdentities(Session::ID_TYPE_EMAIL_PASSWORD)[0] ?? null) {
```

Comment:
I haven't followed recent updates on Shield, but you should be able to just pull `author?->email` and have Shield do the logic in the background, right?

Reply:
You're right. I haven't really used Shield before and it seems I missed the basic functionality. Fixed.